src/main/java/org/wso2/carbon/inbound/amazonsqs/AmazonSQSPollingConsumer.java

```diff
@@ -177,11 +185,23 @@ public Message poll() {
                     if (logger.isDebugEnabled()) {
                         logger.debug("Loading the Content-type : " + contentType + " for " + name);
                     }
+                    msgCtx = this.createMessageContext();
+                    msgCtx.setProperty("MessageId", message.getMessageId());
+                    msgCtx.setProperty("ReceiptHandle", message.getReceiptHandle());
+                    msgCtx.setProperty("MD5OfBody", message.getMD5OfBody());
+
+                    Map<String, MessageAttributeValue> messageAttributes = message.getMessageAttributes();
+                    for (Map.Entry<String, MessageAttributeValue> entry : messageAttributes.entrySet()) {
+                        if (StringUtils.equals(entry.getValue().getDataType(), "Binary")) {
```

Comment:
This can throw nullpointer exception. Also it's a good practise to get the value once and keep in in a method scope variable. getValue is an expensive call.